[Bug]: CMake unnecessarily checks for xaudio2 on Linux

### Version of OpenTTD

15.0.beta1, 14.1 src compile attempted on debian 12.9

### Expected result

After compiling, succed in linking both openttd, openttd-test

### Actual result

from cmake error log
build/OpenTTD-15.0-beta1/build/CMakeFiles/CMakeScratch/TryCompile-psPDzh/src.cxx:3:14: fatal error: windows.h: No such file or directory
    3 |     #include <windows.h>

[CMakeError.log](https://github.com/user-attachments/files/18511951/CMakeError.log)
[CMakeOutput.log](https://github.com/user-attachments/files/18511964/CMakeOutput.log)

searching debian repository shows the most likely candidates include pkgs for ming compiler, libboost and wine.
none of which are mentioned in any docs I could find , and without using closed source.




### Steps to reproduce

D/L openttd-15.0.b1
compile configure using cmake/ccmake
make using make/cmake --build .

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -139,7 +139,9 @@ if(NOT EMSCRIPTEN)
 endif()
 
 if(NOT OPTION_DEDICATED)
-    if(NOT WIN32)
+    if(WIN32)
+        find_package(Xaudio2)
+    else()
         find_package(Allegro)
         if(NOT APPLE)
             find_package(Freetype)
@@ -174,7 +176,6 @@ if(MSVC)
 endif()
 
 find_package(SSE)
-find_package(Xaudio2)
 
 find_package(Grfcodec)
 

